packages/remirror__extension-list/src/list-commands.ts

```diff
@@ -17,10 +18,58 @@ function isList(node: ProsemirrorNode): boolean {
   return !!(
     node.type.spec.group?.includes(ExtensionTag.ListContainerNode) ||
     node.type === schema.nodes.bulletList ||
-    node.type === schema.nodes.orderedList
+    node.type === schema.nodes.orderedList ||
+    node.type === schema.nodes.checkboxList
   );
 }
 
+/**
+ * Checks to see whether this is a list item node.
+ */
+function isListItem(node: ProsemirrorNode): boolean {
+  const schema = node.type.schema;
+
+  return !!(
+    node.type.spec.group?.includes(ExtensionTag.ListItemNode) ||
+    node.type === schema.nodes.listItem ||
+    node.type === schema.nodes.checkboxItem
+  );
+}
+
+function tranformListItems(slice: Slice, itemType: NodeType): ProsemirrorNode[] {
+  const nodes: ProsemirrorNode[] = [];
+
+  slice.content.descendants((node) => {
+    if (isListItem(node) && node.type !== itemType) {
+      const newItem = itemType.createAndFill(node.attrs, node.content);
+
+      if (newItem) {
+        // If we can create a valid node by filling, do that
+        nodes.push(newItem);
+      } else {
+        // Else add in whatever we can
+        let newItem = itemType.create(node.attrs);
+        node.content.forEach((node) => {
+          const frag = Fragment.from(node);
+
+          if (newItem.type.validContent(frag)) {
+            newItem = itemType.create(node.attrs, newItem.content.append(frag));
+          }
+        });
+
+        nodes.push(newItem);
+      }
+
+      return false;
+    }
+
+    nodes.push(node);
+    return false;
+  });
+
+  return nodes;
+}
```

Comment:
I think ideally this should be a function that returns a transaction, containing replace steps that convert a bullet/ordered list to checkbox list (and vice versa), but I really struggled with that to be honest.

Reply:
Also, this should ideally convert nested checklists too